Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/msg/MessageRegistry.java

```diff
@@ -0,0 +1,49 @@
+package edu.asu.conceptpower.rest.msg;
+
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MessageRegistry implements IMessageRegistry {
+
+    @Autowired
+    private ApplicationContext ctx;
+
+    @Autowired
+    private IMessageConverter defaultMessageFactory;
+
+    private Map<String, IMessageConverter> messageConverterMap;
+
+    /**
+     * instantiates all the services under IMessageConverter interface
```

Comment:
what does that mean? Do you mean "all services that implement IMessageConverter"?